src/components/icons/Sort.tsx

```diff
@@ -1,15 +1,36 @@
 import { useSelector } from 'react-redux'
 import IconType from '../../@types/IconType'
 import State from '../../@types/State'
+import ThoughtId from '../../@types/ThoughtId'
 import { HOME_PATH } from '../../constants'
 import getSortPreference from '../../selectors/getSortPreference'
+import getThoughtById from '../../selectors/getThoughtById'
 import head from '../../util/head'
 import AnimatedIcon from './AnimatedIcon'
 import animationData from './animations/10-sort_4.json'
 import animationDataDesc from './animations/11-sort-descanding_7.json'
 
+/** Find the parent ID of a given thought ID. */
+const findParentId = (state: State, thoughtId: ThoughtId): ThoughtId | null => {
+  const thought = getThoughtById(state, thoughtId)
+  return thought ? thought.parentId : null
+}
+
 /** Cursor Sort Direction. */
-const getCursorSortDirection = (state: State) => getSortPreference(state, head(state.cursor || HOME_PATH)).direction
+const getCursorSortDirection = (state: State) => {
+  const cursorId = head(state.cursor || HOME_PATH)
+
+  // Find the parent of cursorId
+  const parentId = findParentId(state, cursorId)
+
+  // Get the sort preference using the parentId
+  const sortPref = getSortPreference(state, parentId!)
```

Comment:
I haven't noticed any runtime errors during testing, but some of the types in this file seem off. My first concern is the non-null assertion operator for `parentId!`. I'd typically rather avoid using assertions based on assumptions that may or may not be true (now, or in the future).

Working backward from there, I don't see why `findParentId` necessarily needs to return `ThoughtId | null` rather than just `ThoughtId`. According to the existing types, `getThoughtById` should always return a non-null `Thought`, and every thought should have a `parentId` -- including the `HOME_PATH` with a `parentId` of itself.

That said, I'm not sure if `getThoughtById` _should_ be guaranteeing that it always returns a non-null `Thought` since it's looking up the given ID in a list which _may_ not contain it. 🤷‍♂️